gmt-4.5.14/src/gmtdefaults.1
.TH GMTDEFAULTS 1 "1 Nov 2015" "GMT 4.5.14" "Generic Mapping Tools"
.SH NAME
gmtdefaults \- To list current \fBGMT\fP defaults
.SH SYNOPSIS
\fBgmtdefaults\fP \fB\-D\fP[\fBu\fP|\fBs\fP] | \fB\-L\fP
.SH DESCRIPTION
\fBgmtdefaults\fP lists the 
\fBGMT\fP parameter defaults if the option \fB\-D\fP is used.  There are three ways to change some of the settings:
(1) Use the command \fBgmtset\fP, (2) use any texteditor to edit the file \.gmtdefaults4 in your home, ~/.gmt
or current directory (if you do not have this file, run 
\fBgmtdefaults\fP -D > ~/.gmtdefaults4 to get one with the system default settings), or (3) override any parameter
by specifying one or more \fB\-\-PARAMETER\fP=\fIvalue\fP statements on the commandline of any \fBGMT\fP command
(\fBPARAMETER\fP and \fBVALUE\fP are any combination listed below).  The
first two options are permanent changes until explicitly changed back, while the last option is ephemeral
and only applies to the single \fBGMT\fP command that received the override.
\fBGMT\fP can provide default values in US or SI units.  This choice is determined by the contents of the
gmt_setup.conf file in \fBGMT\fP's share directory.\"'
.TP
\fB\-D\fP
Print the system \fBGMT\fP defaults to standard output.  Append \fBu\fP for US
defaults or \fBs\fP for SI defaults. [\fB\-D\fP alone gives current choice in gmt_setup.conf].
.TP
\fB\-L\fP
Print the user's currently active defaults to standard output.\"'
.br
.sp
Your currently active defaults come from the \.gmtdefaults4 file in the 
current working directory, if present; else from the \.gmtdefaults4 file 
in your home directory, if present; else from the file ~/.gmt/.gmtdefaults4, if present;
else from the system defaults set at the time \fBGMT\fP was compiled.
.SH \fBGMT\fP PARAMETERS
The following is a list of the parameters that are user-definable in \fBGMT\fP.  The
parameter names are always given in UPPER CASE.  The parameter values are
case-insensitive unless otherwise noted.  The
system defaults are given in brackets [ for SI (and US) ].  Those marked \fB*\fP can be set on the command
line as well (the corresponding option is given in parentheses).  Note that default distances and
lengths below are given in both cm or inch; the chosen default depends on your choice of default
unit (see \fBMEASURE_UNIT\fP).  You can explicitly specify the unit used for distances and
lengths by appending \fBc\fP (cm), \fBi\fP (inch), \fBm\fP (meter), or \fBp\fP (points).
When no unit is indicated the value will be assumed to be in the unit set by \fBMEASURE_UNIT\fP.
Note that the printer resolution \fBDOTS_PR_INCH\fP is always the number of dots or pixels per inch.
Several parameters take only TRUE or FALSE.
.TP
.B ANNOT_FONT_PRIMARY
Font used for upper annotations, etc. [Helvetica].  Specify either the font number or
the font name (case sensitive!).  The 35 available fonts are:
.sp
.br
0	Helvetica
.br
1	Helvetica-Bold
.br
2	Helvetica-Oblique
.br
3	Helvetica-BoldOblique
.br
4	Times-Roman
.br
5	Times-Bold
.br
6	Times-Italic
.br
7	Times-BoldItalic
.br
8	Courier
.br
9	Courier-Bold
.br
10	Courier-Oblique
.br
11	Courier-BoldOblique
.br
12	Symbol
.br
13	AvantGarde-Book
.br
14	AvantGarde-BookOblique
.br
15	AvantGarde-Demi
.br
16	AvantGarde-DemiOblique
.br
17	Bookman-Demi
.br
18	Bookman-DemiItalic
.br
19	Bookman-Light
.br
20	Bookman-LightItalic
.br
21	Helvetica-Narrow
.br
22	Helvetica-Narrow-Bold
.br
23	Helvetica-Narrow-Oblique
.br
24	Helvetica-Narrow-BoldOblique
.br
25	NewCenturySchlbk-Roman
.br
26	NewCenturySchlbk-Italic
.br
27	NewCenturySchlbk-Bold
.br
28	NewCenturySchlbk-BoldItalic
.br
29	Palatino-Roman
.br
30	Palatino-Italic
.br
31	Palatino-Bold
.br
32	Palatino-BoldItalic
.br
33	ZapfChancery-MediumItalic
.br
34	ZapfDingbats
.TP
.B ANNOT_FONT_SIZE_PRIMARY
Font size (> 0) for map annotations  [14p].
.TP
.B ANNOT_FONT_SECONDARY
Font to use for time axis secondary annotations.  See \fBANNOT_FONT_PRIMARY\fP for available fonts  [Helvetica].
.TP
.B ANNOT_FONT_SIZE_SECONDARY
Font size (> 0) for time axis secondary annotations  [16p].
.TP
.B ANNOT_MIN_ANGLE
If the angle between the map boundary and the annotation baseline is less than this minimum value (in degrees), the
annotation is not plotted (this may occur for certain oblique projections.)  Give a value in the
range 0\-90. [20]
.TP
.B ANNOT_MIN_SPACING
If an annotation would be plotted less than this minimum distance from its closest neighbor, the
annotation is not plotted (this may occur for certain oblique projections.) [0]
.TP
.B ANNOT_OFFSET_PRIMARY
Distance from end of tickmark to start of annotation [0.2\fBc\fP (or 0.075\fBi\fP)].  A negative
offset will place the annotation inside the map border.
.TP
.B ANNOT_OFFSET_SECONDARY
Distance from base of primary annotation to the top of the secondary annotation [0.2\fBc\fP (or 0.075\fBi\fP)] (Only
applies to time axes with both primary and secondary annotations).
.TP
.B BASEMAP_AXES
Sets which axes to draw and annotate.  Case sensitive: Upper case means both draw and annotate, lower case
means draw axis only.  [WESN].
.TP
.B BASEMAP_FRAME_RGB
Color used to draw map boundaries and annotations.
Give a red/green/blue triplet, with each element in the 0\-255 range.
Prepend '+' to replicate this color to the tick-, grid-, and frame-pens.
[0/0/0] (black).
.TP
.B BASEMAP_TYPE
Choose between inside, graph, plain and fancy (thick boundary, alternating black/white frame; append + for rounded corners)  [fancy].
For some map projections (e.g., Oblique Mercator), plain is the only option even if fancy is set as default.
In general, fancy only applies to situations where the projected x and y directions parallel the lon and lat
directions (e.g., rectangular projections, polar projections).  For situations where all boundary ticks and annotations
must be inside the maps (e.g., for preparing geotiffs), chose inside. Finally, graph is used for linear projections only
and will extend the axis by 7.5% and add arrow heads.
.TP
.B CHAR_ENCODING
Names the eight bit character set being used for text in files and in command line parameters. This allows \fBGMT\fP
to ensure that the \fIPostScript\fP output generates the correct characters on the plot..  Choose from Standard, Standard+, ISOLatin1, ISOLatin1+,
and ISO-8859-x (where x is in the ranges 1-10 or 13-15).  See Appendix F for details [ISOLatin1+ (or Standard+)].
.TP
.B COLOR_BACKGROUND
Color used for the background of images (i.e., when z < lowest colortable entry).
Give a red/green/blue triplet, with each element in the 0\-255 range. [0/0/0] (black)
.TP
.B COLOR_FOREGROUND
Color used for the foreground of images (i.e., when z > highest colortable entry).
Give a red/green/blue triplet, with each element in the 0\-255 range. [255/255/255] (white)
.TP
.B COLOR_IMAGE
Selects which operator to use when rendering bit-mapped color images.  Due to the
lack of the colorimage operator in some \fIPostScript\fP implementations, as well as
some \fIPostScript\fP editors inability to handle color gradations, \fBGMT\fP offers
two different options:
.br
.sp
	adobe (Adobe's colorimage definition)  [Default].\"'
.br
	tiles (Plot image as many individual rectangles).
.br
.TP
.B COLOR_MODEL
Selects if color palette files contain RGB values (r,g,b in 0-255 range), HSV values
(h in 0-360, s,v in 0-1 range), or CMYK values (c,m,y,k in 0-1 range).  A \fBCOLOR_MODEL\fP setting in the color palette file
will override this setting.  Internally, color interpolation takes place directly
on the RGB values which can give unexpected hues, whereas interpolation directly on the
HSV values better preserves the hues.  Prepend the prefix "+" to force interpolation in
the selected color system (does not apply to the CMYK system). For this additional option, the defaults take
precedence over the color palette file  [rgb].
.TP
.B COLOR_NAN
Color used for the non-defined areas of images (i.e., where z == NaN).
Give a red/green/blue triplet, with each element in the 0\-255 range. [128/128/128] (gray)
.TP
.B D_FORMAT
Output format (C language printf syntax) to be used when printing double precision floating point numbers.
For geographic coordinates, see \fBOUTPUT_DEGREE_FORMAT\fP. [%.12g].
.TP
.B DEGREE_SYMBOL
Determines what symbol is used to plot the degree symbol on geographic map annotations.  Choose between
ring, degree, colon, or none [ring].
.TP
.B DOTS_PR_INCH
Resolution of the plotting device (dpi).  Note that in order to be as compact as possible,
\fBGMT\fP \fIPostScript\fP output uses integer formats only so the resolution should be
set depending on what output device you are using.  E.g, using 300 and sending the output to
a Linotype 300 phototypesetter (2470 dpi) will not take advantage of the extra resolution (i.e.,
positioning on the page and line thicknesses are still only done in steps of 1/300 inch; of course, text will look smoother)  [300].
.TP
.B ELLIPSOID
The (case sensitive) name of the ellipsoid used for the map projections  [WGS-84].  Choose among:
.sp
.br
WGS-84 : World Geodetic System [Default] (1984)
.br
OSU91A : Ohio State University (1991)
.br
OSU86F : Ohio State University (1986)
.br
Engelis : Goddard Earth Models (1985)
.br
SGS-85 : Soviet Geodetic System (1985)
.br
TOPEX : Used commonly for altimetry (1990)
.br
MERIT-83 : United States Naval Observatory (1983)
.br
GRS-80 : International Geodetic Reference System (1980)
.br
Hughes-1980 : Hughes Aircraft Company for DMSP SSM/I grid products (1980)
.br
Lerch : For geoid modelling (1979)
.br
ATS77 : Average Terrestrial System, Canada Maritime provinces (1977)
.br
IAG-75 : International Association of Geodesy (1975)
.br
Indonesian : Applies to Indonesia (1974)
.br
WGS-72 : World Geodetic System (1972)
.br
NWL-10D : Naval Weapons Lab (Same as WGS-72) (1972)
.br
South-American : Applies to South America (1969)
.br
Fischer-1968 : Used by NASA for Mercury program (1968)
.br
Modified-Mercury-1968 : Same as Fischer-1968 (1968)
.br
GRS-67 : International Geodetic Reference System (1967)
.br
International-1967 : Worldwide use (1967)
.br
WGS-66 : World Geodetic System (1966)
.br
NWL-9D : Naval Weapons Lab (Same as WGS-66) (1966)
.br
Australian : Applies to Australia (1965)
.br
APL4.9 : Appl. Physics (1965)
.br
Kaula : From satellite tracking (1961)
.br
Hough : Applies to the Marshall Islands (1960)
.br
WGS-60 : World Geodetic System (1960)
.br
Fischer-1960 : Used by NASA for Mercury program (1960)
.br
Mercury-1960 : Same as Fischer-1960 (1960)
.br
Modified-Fischer-1960 : Applies to Singapore (1960)
.br
Fischer-1960-SouthAsia : Same as Modified-Fischer-1960 (1960)
.br
Krassovsky : Used in the (now former) Soviet Union (1940)
.br
War-Office : Developed by G. T. McCaw (1926)
.br
International-1924 : Worldwide use (1924)
.br
Hayford-1909 : Same as the International 1924 (1909)
.br
Helmert-1906 : Applies to Egypt (1906)
.br
Clarke-1880 : Applies to most of Africa, France (1880)
.br
Clarke-1880-Arc1950 : Modified Clarke-1880 for Arc 1950 (1880)
.br
Clarke-1880-IGN : Modified Clarke-1880 for IGN (1880)
.br
Clarke-1880-Jamaica : Modified Clarke-1880 for Jamaica (1880)
.br
Clarke-1880-Merchich : Modified Clarke-1880 for Merchich (1880)
.br
Clarke-1880-Palestine : Modified Clarke-1880 for Palestine (1880)
.br
Andrae : Applies to Denmark and Iceland (1876)
.br
Clarke-1866 : Applies to North America, the Philippines (1866)
.br
Clarke-1866-Michigan : Modified Clarke-1866 for Michigan (1866)
.br
Struve : Friedrich Georg Wilhelm Struve (1860)
.br
Clarke-1858 : Clarke's early ellipsoid (1858)
.br
Airy : Applies to Great Britain (1830)
.br
Airy-Ireland : Applies to Ireland in 1965 (1830)
.br
Modified-Airy : Same as Airy-Ireland (1830)
.br
Bessel : Applies to Central Europe, Chile, Indonesia (1841)
.br
Bessel-Schwazeck : Applies to Namibia (1841)
.br
Bessel-Namibia : Same as Bessel-Schwazeck (1841)
.br
Bessel-NGO1948 : Modified Bessel for NGO 1948 (1841)
.br
Everest-1830 : India, Burma, Pakistan, Afghanistan, Thailand (1830)
.br
Everest-1830-Kalianpur : Modified Everest for Kalianpur (1956) (1830)
.br
Everest-1830-Kertau : Modified Everest for Kertau, Malaysia & Singapore (1830)
.br
Everest-1830-Timbalai : Modified Everest for Timbalai, Sabah Sarawak (1830)
.br
Everest-1830-Pakistan : Modified Everest for Pakistan (1830)
.br
Walbeck : First least squares solution by Finnish astronomer (1819)
.br
Plessis : Old ellipsoid used in France (1817)
.br
Delambre : Applies to Belgium (1810)
.br
CPM : Comm. des Poids et Mesures, France (1799)
.br
Maupertius : Really old ellipsoid used in France (1738)
.br
Sphere : The mean radius in WGS-84 (for spherical/plate tectonics applications) (1984)
.br
Moon : Moon (IAU2000) (2000)
.br
Mercury : Mercury (IAU2000) (2000)
.br
Venus : Venus (IAU2000) (2000)
.br
Mars : Mars (IAU2000) (2000)
.br
Jupiter : Jupiter (IAU2000) (2000)
.br
Saturn : Saturn (IAU2000) (2000)
.br
Uranus : Uranus (IAU2000) (2000)
.br
Neptune : Neptune (IAU2000) (2000)
.br
Pluto : Pluto (IAU2000) (2000)
.sp
Note that for some global projections, \fBGMT\fP may use a spherical approximation of the
ellipsoid chosen, setting the flattening to zero, and using a mean radius.
A warning will be given when this happens.
If a different
ellipsoid name than those mentioned here is given, \fBGMT\fP will attempt to parse the
name to extract the semi-major axis (\fIa\fP in m) and the flattening. Formats allowed are:
.br
.sp
	\fIa\fP		implies a zero flattening
.br
	\fIa\fP,\fIinv_f\fP	where \fIinv_f\fP is the inverse flattening
.br
	\fIa\fP,\fBb=\fP\fIb\fP	where \fIb\fP is the semi-minor axis (in m)
.br
	\fIa\fP,\fBf=\fP\fIf\fP	where \fIf\fP is the flattening
.br
.sp
This way a
custom ellipsoid (e.g., those used for other planets) may be used. Further note that coordinate
transformations in \fBmapproject\fP can also specify specific datums; see the \fBmapproject\fP
man page for further details and how to view ellipsoid and datum parameters.
.TP
.B FIELD_DELIMITER
This setting determines what character will separate ASCII output data columns written by \fBGMT\fP.  Choose
from tab, space, comma, and none [tab].
.TP
.B FRAME_PEN
Pen attributes used to draw plain map frame in dpi units or points (append p)  [1.25p].
.TP
.B FRAME_WIDTH
Width (> 0) of map borders for fancy map frame [0.2\fBc\fP (or 0.075\fBi\fP)].
.TP
.B GLOBAL_X_SCALE
Global x-scale (> 0) to apply to plot-coordinates before plotting.  Normally used to shrink
the entire output down to fit a specific height/width  [1.0].
.TP
.B GLOBAL_Y_SCALE
Same, but for y-coordinates  [1.0].
.TP
.B GRID_CROSS_SIZE_PRIMARY
Size (>= 0) of grid cross at lon-lat intersections.  0 means draw continuous gridlines
instead [0].
.TP
.B GRID_CROSS_SIZE_SECONDARY
Size (>= 0) of grid cross at secondary lon-lat intersections.  0 means draw continuous gridlines
instead [0].
.TP
.B GRID_PEN_PRIMARY
Pen attributes used to draw grid lines in dpi units or points (append p) [0.25p].
.TP
.B GRID_PEN_SECONDARY
Pen attributes used to draw grid lines in dpi units or points (append p) [0.5p].
.TP
.B GRIDFILE_FORMAT
Default file format for grids, with optional scale, offset and invalid value, written as
\fIff\fP/\fIscale\fP/\fIoffset\fP/\fIinvalid\fP. The 2-letter format indicator can be one of [\fBbcnsr\fP][\fBbsifd\fP]. The first
letter indicates native \fBGMT\fP binary, old format netCDF, COARDS-compliant netCDF, Surfer format or Sun
Raster format. The second letter stands for byte, short, int, float and double, respectively. When
/\fIinvalid\fP is omitted the appropriate value for the given format is used (NaN or largest negative).
When /\fIscale\fP/\fIoffset\fP is omitted, /1.0/0.0 is used. [nf].
.TP
.B GRIDFILE_SHORTHAND
If TRUE, all grid file names are examined to see if they use the file extension
shorthand discussed in Section 4.17 of the \fBGMT\fP Technical Reference and Cookbook.
If FALSE, no filename expansion is done [FALSE].
.TP
.B HEADER_FONT
Font to use when plotting headers.  See \fBANNOT_FONT_PRIMARY\fP for available fonts  [Helvetica].
.TP
.B HEADER_FONT_SIZE
Font size (> 0) for header  [36p].
.TP
.B HEADER_OFFSET
Distance from top of axis annotations (or axis label, if present) to base of plot header [0.5\fBc\fP (or 0.1875\fBi\fP)].
.TP
.B HISTORY
If TRUE, passes the history of past common command options via the hidden .gmtcommands4 file [TRUE].
.TP
.B HSV_MAX_SATURATION
Maximum saturation (0\-1) assigned for most positive intensity value [0.1].
.TP
.B HSV_MIN_SATURATION
Minimum saturation (0\-1) assigned for most negative intensity value [1.0].
.TP
.B HSV_MAX_VALUE
Maximum value (0\-1) assigned for most positive intensity value [1.0].
.TP
.B HSV_MIN_VALUE
Minimum value (0\-1) assigned for most negative intensity value [0.3].
.TP
.B INPUT_CLOCK_FORMAT
Formatting template that indicates how an input clock string is formatted.  This template
is then used to guide the reading of clock strings in data fields.  To properly decode
12-hour clocks, append am or pm (or upper case) to match your data records. As examples,
try hh:mm, hh:mm:ssAM, etc. [hh:mm:ss].
.TP
.B INPUT_DATE_FORMAT
Formatting template that indicates how an input date string is formatted.  This template
is then used to guide the reading of date strings in data fields.  You may specify either
Gregorian calendar format or ISO week calendar format.  Gregorian calendar: Use any
combination of yyyy (or yy for 2-digit years; if so see \fBY2K_OFFSET_YEAR\fP), mm (or o
for abbreviated month name in the current time language),
and dd, with or without delimiters.  For day-of-year data, use jjj instead of mm and/or dd.
Examples can be ddmmyyyy, yy-mm-dd, dd-o-yyyy, yyyy/dd/mm, yyyy-jjj, etc.
ISO Calendar:  Expected template is yyyy[-]W[-]ww[-]d, where ww is ISO week and d is ISO week
day.  Either template must be consistent, e.g., you cannot specify months if you don't specify\"'
years.  Examples are yyyyWwwd, yyyy-Www, etc. [yyyy-mm-dd].
.TP
.B INTERPOLANT
Determines if linear (linear), Akima's spline (akima), natural cubic spline (cubic) or\"'
no interpolation (none) should be used for 1-D interpolations in various programs [akima].
.TP
.B IO_HEADER
(\fB* \-H\fP) Specifies whether input/output ASCII files have header record(s) or not  [FALSE].
.TP
.B LABEL_FONT
Font to use when plotting labels below axes.  See \fBANNOT_FONT_PRIMARY\fP for available fonts  [Helvetica].
.TP
.B LABEL_FONT_SIZE
Font size (> 0) for labels  [24p].
.TP
.B LABEL_OFFSET
Distance from base of axis annotations to the top of the axis label [0.3\fBc\fP (or 0.1125\fBi\fP)].
.TP
.B LINE_STEP
Determines the maximum length (> 0) of individual straight line-segments when drawing arcuate lines  [0.025\fBc\fP (or 0.01\fBi\fP)]
.TP
.B MAP_SCALE_FACTOR
Changes the default map scale factor used for the Polar Stereographic [0.9996], UTM [0.9996], and Transverse Mercator [1] projections
in order to minimize areal distortion.  Provide a new scale-factor or leave as default.
.TP
.B MAP_SCALE_HEIGHT
Sets the height (> 0) on the map of the map scale bars drawn by various programs [0.2\fBc\fP (or 0.075\fBi\fP)].
.TP
.B MEASURE_UNIT
Sets the unit length.  Choose between cm, inch, m, and point.  [cm].  Note
that, in \fBGMT\fP, one point is defined as 1/72 inch (the \fIPostScript\fP definition), while it
is often defined as 1/72.27 inch in the typesetting industry.  There is no
universal definition.
.TP
.B N_COPIES
(\fB* \-c\fP) Number of plot copies to make  [1].
.TP
.B N_HEADER_RECS
Specifies how many header records to expect if \fB\-H\fP is turned on  [1].
.TP
.B NAN_RECORDS
Determines what happens when input records containing NaNs for \fIx\fP or \fIy\fP (and in some
cases \fIz\fP) are read.  Choose between \fIskip\fP, which will simply report how many bad
records were skipped, and \fIpass\fP [Default], which will pass these records on to the calling programs.
For most programs this will result in output records with NaNs as well, but some will interpret these
NaN records to indicate gaps in a series; programs may then use that information to detect segmentation (if
applicable).
.TP
.B OBLIQUE_ANNOTATION
This integer is a sum of 6 bit flags (most of which only are relevant for oblique projections):
If bit 1 is set (1), annotations will occur wherever a gridline crosses the
map boundaries, else longitudes will be annotated on the lower and upper boundaries only, and latitudes will be
annotated on the left and right boundaries only.  If bit 2 is set (2), then longitude annotations will
be plotted horizontally.  If bit 3 is set (4), then latitude annotations will be plotted horizontally.
If bit 4 is set (8), then oblique tickmarks are extended to give a projection equal to the specified tick_length.
If bit 5 is set (16), tickmarks will be drawn normal to the border regardless of gridline angle.
If bit 6 is set (32), then latitude annotations will be plotted parallel to the border.
To set a combination of these, add up the values in parentheses. [1].
.TP
.B OUTPUT_CLOCK_FORMAT
Formatting template that indicates how an output clock string is to be formatted.  This template
is then used to guide the writing of clock strings in data fields.  To use a floating point
format for the smallest unit (e.g., seconds), append .xxx, where the number of x indicates
the desired precision.  If no floating point is indicated then the smallest specified unit
will be rounded off to nearest integer.  For 12-hour clocks, append am, AM, a.m., or A.M. (\fBGMT\fP
will replace a|A with p|P for pm).  If your template starts with a leading hyphen (-) then each integer item (y,m,d) will be
printed without leading zeros (default uses fixed width formats).  As examples, try
hh:mm, hh.mm.ss, hh:mm:ss.xxxx, hha.m., etc. [hh:mm:ss].
.TP
.B OUTPUT_DATE_FORMAT
Formatting template that indicates how an output date string is to be formatted.  This template
is then used to guide the writing of date strings in data fields.  You may specify either
Gregorian calendar format or ISO week calendar format.  Gregorian calendar: Use any
combination of yyyy (or yy for 2-digit years; if so see \fBY2K_OFFSET_YEAR\fP), mm (or o for
abbreviated month name in the current time language),
and dd, with or without delimiters.  For day-of-year data, use jjj instead of mm and/or dd.
As examples, try yy/mm/dd, yyyy=jjj, dd-o-yyyy, dd-mm-yy, yy-mm, etc.
ISO Calendar:  Expected template is yyyy[-]W[-]ww[-]d, where ww is ISO week and d is ISO week
day.  Either template must be consistant, e.g., you cannot specify months if you don't specify\"'

gmt-4.5.14/ChangeLog
-----------------------------------------------------------------------------------------
	$Id: ChangeLog 10331 2015-10-29 19:41:56Z pwessel $
-----------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------
2015-11-01	pwessel
	* GMT 4.5.14 finalized and released
-----------------------------------------------------------------------------------------
2015-10-29	pwessel
	* meca/psmeca.c:	Enforce rakes and strikes to be in -180/+180 range.
				Issue #799, for gmt4 only as not a problem in gmt5.

2015-07-28	pwessel
	* meca/psmeca.c:	Be able to read any value x and y coordinates

2015-06-24	pwessel
	* gmt_support.c:	Fix bug in jump near Dateline in gmt_inonout_sphpol_count,
				affecting inside/outside spherical cases, reported in message-2219

2015-03-18	pwessel
	* share/cpt/GMT_wysiwyg.cpt:	Fix typo in blue for high end [Issue #689].

2015-01-16	pwessel
	* gmt_plot.c:		Fix issue # 662 with graph arrows for negative scales.

2015-01-10	pwessel
	* gmt_plot.c:		Flipping the n/s sides array when lon and -y axes
				fixes issue #520

2015-01-01	pwessel
	* configure.ac:		Moved to 4.5.14 and back to svn release only.
-----------------------------------------------------------------------------------------
2015-01-01	pwessel
	* GMT 4.5.13 finalized and released [GSHHG 2.3.4 was released on Jan 1 as well]
-----------------------------------------------------------------------------------------
2014-12-28	pwessel
	* gmt_map.c:		Needed separate GMT_cartesian_overlap for non-geographic
				projections.  Fixes issue # 657.

2014-09-29	pwessel
	* gmt_plot.c:		Left a PSL variable undefined when grdimage plotted a grid
				with x = longitude and y = Cartesian.

2014-09-09	pwessel
	* grdfilter.c:		Fix y-shifts when -I sets smaller output.  Issue #616.
	* gmt_bcr.c:		Fix problem with points slightly outside causing an
				access violation; see issue #620.

2014-08-29	pwessel
	* sample1d.c:		Fix resampling for decreasing t values.  Issue #608.

2014-08-05	pwessel
	* gmt_map.c:		Could run into trouble clipping polygons for non-periodic
				maps.  Added more checks in GMT_wesn_clip.

2014-07-31	pwessel
	* gmt_map.c:		Rect clipping with -R<w/s/e/n>r could introduce stray lines
				when dataset has features about 180 degrees away from
				the user's area. See issue # 594.

2014-07-30	pwessel
	* configure.ac:		Set default GSHHG version to 2.3.2 (released August 1, 2014)
	* psxy[z].c:		Fixed wrap-around issue in fault lines.

2014-07-27	pwessel
	* gmtselect.c:		The -Z option needed to pass records with NaNs.

2014-07-19	pwessel
	* x2sys/x2sys_solve.c:	Fix problems that arise when a track has less crossovers
				than model parameters.
	* gmtmath.c:		Fix issue #123 with files and -C.

2014-07-01	pwessel
	* configure.ac:		Set default GSHHG version to 2.3.1 (released July 1, 2014)
	
2014-06-28	pwessel
	* mgd77/mgd77sniffer.c:	Needed to check that grid values used in decimate function
	 			also were within reasonable range.

2014-05-26	pwessel
	* gmt_map.c:		The map region clip path for -JE/e was hardwired to a full
				circle and did not care about -R setting.
				When -R...r and -Je/E was given the search for enclosing
				geographic boundaries failed when either the S or N pole
				was the projection's antipole.
2014-05-25	pwessel
	* grdmath.c:		Fix issue #566 bug in grdmath's KM2DEG operator.

2014-03-19	pwessel
	* gmtselect.c:		The -Fpolygon option might miss points because the io-machinery
				would sometimes set the min/max longitudes found incorrectly.

2014-03-14	pwessel
	* gmt_shore.c:		Give more error feedback if nc_open fails due to old library.

2014-03-01	pwessel
	* configure.ac:		Moved to 4.5.13 and back to svn release only.

-----------------------------------------------------------------------------------------
2014-03-01	pwessel
	* GMT 4.5.12 finalized and released [GSHHG 2.3.0 was released on Feb 1]
-----------------------------------------------------------------------------------------
2014-02-17	fwobbe
	* gmt_init.c etc.:	File locking -- if enabled via -DFLOCK -- is
				non-compulsory now.  This means if fcntl fails for some reason,
				GMT continues to access the file.  Hence, it is recommended to
				configure GMT with --enable-flock even if fcntl might be buggy
				on some ancient NFS shares.

2014-02-17	pwessel
	* gmt_map.c:		Size for memory allocation cast the product of nx * ny to
				size_t but overflow had already happen as calculation was
				done using ints, then cast to size_t. Must cast nx and ny
				separately.
2014-02-10	jluis
	* misc/gmt2kml.c:	Make sure the default style IDs are unique for each process.

2014-02-07	jluis
	* ps2raster.c:		Make sure that images in KML files do not carry a directory name

2014-02-03	pwessel
	* pslib.c:		Auto-widen paragraph width to at least fit widest word.

2014-02-01	pwessel
	* configure.ac:		Since GSHHG 2.3.0 was released today, make that the default

2014-01-26	pwessel
	* gmt_grdio.c:		Building the grid command string could exceed its length if
				the argument became exactly the max length of the array; then
				a final extra character \0 would be added and exceed array length.
				Thanks to Joachim Saul for pointing it out.

2014-01-22	pwessel
	* pscoast.c:		The processing of -I -N levels and pens could get out of sync
				so that the wrong pen was used for the specified feature.

2014-01-16	remko
	* pslib.c:		Take superscripts and subscripts into account when determining
				dimensions of text box.

2014-01-10	pwessel
	* grdmath.c:		Had to let SDIST return km instead of degrees since it cannot
				do degrees for geodesics.  By doing km it is now aligned with
				what PDIST and LDIST returns for geographic data, and matches
				what GMT5 does.  Some scripts had to be modified to handle the
				new behavior.  If you use SDIST you may wish to check your usage.
				Also fixed bug in SDIST in that spherical selected geodesics

gmt-4.5.14/src/gmtdefaults.c
/*--------------------------------------------------------------------
 *	$Id: gmtdefaults.c 10289 2014-12-28 21:17:06Z pwessel $
 *
 *	Copyright (c) 1991-2015 by P. Wessel and W. H. F. Smith
 *	See LICENSE.TXT file for copying and redistribution conditions.
 *
 *	This program is free software; you can redistribute it and/or modify
 *	it under the terms of the GNU General Public License as published by
 *	the Free Software Foundation; version 2 or any later version.
 *
 *	This program is distributed in the hope that it will be useful,
 *	but WITHOUT ANY WARRANTY; without even the implied warranty of
 *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *	GNU General Public License for more details.
 *
 *	Contact info: gmt.soest.hawaii.edu
 *--------------------------------------------------------------------*/
/*
 * gmtdefaults will list the users default settings for the GMT-SYSTEM or
 * (by using the -D option), get the GMT-SYSTEM's default settings.
 *
 * Author:	Paul Wessel
 * Date:	12-JUL-2000
 * Version:	4.0
 */
 
#define GMT_WITH_NO_PS
#include "gmt.h"

int main (int argc, char **argv)
{
	GMT_LONG i, get = 0;

	GMT_LONG error = FALSE, get_sys_defaults = FALSE, get_user_defaults = FALSE;

	char *path = NULL;

	/* SPECIAL INITIALIZATION SINCE BMT_begin IS NOT USED HERE !! */
#ifdef DEBUG
	GMT_memtrack_init (&GMT_mem_keeper);
#endif
	for (i = strlen(argv[0]); i >= 0 && argv[0][i] != '/'; i--);
	GMT_program = &argv[0][i+1];	/* Name without full path */
	GMT_set_home ();
	GMT_init_fonts (&GMT_N_FONTS);

	GMT_begin_io ();

	for (i = 1; !error && i < argc; i++) {
		if (argv[i][0] != '-') continue;
		switch (argv[i][1]) {
			case '\0':
				error += GMT_parse_common_options (argv[i], 0, 0, 0, 0);
				break;
			case 'D':	/* Get GMT defaults settings */
				get_sys_defaults = TRUE;
				switch (argv[i][2]) {
					case 'S':	/* SI version */
					case 's':
						get = 1;
						break;
					case 'U':	/* US version */
					case 'u':
						get = 2;
						break;
					default:	/* Version chosen in gmt_setup.conf */
						get = 0;
						break;
				}
				break;
			case 'L':	/* List the user's current GMT defaults settings */
				get_user_defaults = TRUE;
				break;
			default:
				error = TRUE;
                                GMT_default_error (argv[i][1]);
                                exit (EXIT_FAILURE);
				break;
		}
	}

	if (argc == 1 || GMT_give_synopsis_and_exit) {
		fprintf (stderr, "gmtdefaults %s - List GMT default parameters to stdout\n\n", GMT_VERSION);
		fprintf (stderr, "usage: gmtdefaults [-D[s|u] | -L]\n\n");
		if (GMT_give_synopsis_and_exit) exit (EXIT_FAILURE);
		fprintf (stderr, "\t-D prints the default settings for the GMT system.\n");
		fprintf (stderr, "\t   Append s to see the SI version of defaults.\n");
		fprintf (stderr, "\t   Append u to see the US version of defaults.\n");
		fprintf (stderr, "\t-L prints the users current GMT default settings.\n");
		exit (EXIT_FAILURE);
	}

	if ((get_user_defaults + get_sys_defaults) != 1) {
		fprintf (stderr, "%s: GMT SYNTAX ERROR: Must specify one of -D or -L\n", GMT_program);
		error++;
	}

	if (error) exit (EXIT_FAILURE);

	if (get_user_defaults) {
		GMT_getdefaults (CNULL);
	}
	else {
		GMT_getdefpath (get, &path);
		GMT_getdefaults (path);
#ifdef WIN32
		/* We have a "race" condition here. Plain free() crashes Windows but *nix doesn't like GMT_free */
		GMT_free ((void *)path);
#else
		free ((void *)path);
#endif
	}

	GMT_putdefaults ("-");

	GMT_end (argc, argv);

	exit (EXIT_SUCCESS);
}
